ui/admin/app/components/form/target/details/index.hbs

```diff
@@ -75,86 +81,141 @@
+    @isOptional={{true}}
     @value={{@model.session_max_seconds}}
-    @label={{t 'form.session_max_seconds.label'}}
-    @error={{@model.errors.session_max_seconds}}
-    @helperText={{t 'form.session_max_seconds.help'}}
-    min='0'
-    step='1'
-    as |field|
+    @isInvalid={{@model.errors.session_max_seconds}}
+    @type='text'
+    name='session_max_seconds'
+    disabled={{form.disabled}}
+    {{on 'input' (set-from-event @model 'session_max_seconds')}}
+    as |F|
   >
+    <F.Label>{{t 'form.session_max_seconds.label'}}</F.Label>
+    <F.HelperText>{{t 'form.session_max_seconds.help'}}</F.HelperText>
     {{#if @model.errors.session_max_seconds}}
-      <field.errors as |errors|>
+      <F.Error as |E|>
         {{#each @model.errors.session_max_seconds as |error|}}
-          <errors.message>{{error.message}}</errors.message>
+          <E.Message>{{error.message}}</E.Message>
         {{/each}}
-      </field.errors>
+      </F.Error>
     {{/if}}
-  </form.input>
+  </Hds::Form::TextInput::Field>
 
-  <form.input
-    @name='session_connection_limit'
-    @type='number'
+  <Hds::Form::TextInput::Field
+    @isRequired={{false}}
+    @isOptional={{true}}
     @value={{@model.session_connection_limit}}
-    @label={{t 'form.session_connection_limit.label'}}
-    @error={{@model.errors.session_connection_limit}}
-    @helperText={{t 'form.session_connection_limit.help'}}
-    min='-1'
-    step='1'
-    as |field|
+    @isInvalid={{@model.errors.session_connection_limit}}
+    @type='text'
+    name='session_connection_limit'
+    disabled={{form.disabled}}
+    {{on 'input' (set-from-event @model 'session_connection_limit')}}
+    as |F|
   >
+    <F.Label>{{t 'form.session_connection_limit.label'}}</F.Label>
+    <F.HelperText>{{t 'form.session_connection_limit.help'}}</F.HelperText>
     {{#if @model.errors.session_connection_limit}}
-      <field.errors as |errors|>
+      <F.Error as |E|>
         {{#each @model.errors.session_connection_limit as |error|}}
-          <errors.message>{{error.message}}</errors.message>
-        {{/each}}
-      </field.errors>
-    {{/if}}
-  </form.input>
-
-  <form.input
-    @name='worker_filter'
-    @type='text'
-    @value={{@model.worker_filter}}
-    @label={{t 'form.worker_filter.label'}}
-    @error={{@model.errors.worker_filter}}
-    @helperText={{t 'form.worker_filter.help'}}
-    @linkText={{t 'actions.learn-more'}}
-    @link={{doc-url 'target.worker-filters'}}
-    as |field|
-  >
-    {{#if @model.errors.worker_filter}}
-      <field.errors as |errors|>
-        {{#each @model.errors.worker_filter as |error|}}
-          <errors.message>{{error.message}}</errors.message>
+          <E.Message>{{error.message}}</E.Message>
         {{/each}}
-      </field.errors>
+      </F.Error>
     {{/if}}
-  </form.input>
+  </Hds::Form::TextInput::Field>
 
-  <form.input
-    @name='default_port'
-    @type='number'
+  <Hds::Form::TextInput::Field
+    @isRequired={{false}}
```

Comment:
Isn't the default port required?

Reply:
Funny enough, yes and no. For SSH Targets it is not required but it is for TCP. I marked it as required again.